### PR TITLE
chore(codegen): bump smithy-typescript version to 0.21.1

### PR DESCRIPTION
**Pull request by kstich**
Code generation diff: https://github.com/aws/aws-sdk-js-v3/commit/b471e06523da158ad158d39f35c0c95a0eb14157

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: scripts/generate-clients/config.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Update this commit when taking up new changes from smithy-typescript.
 module.exports = {
   // Use full commit hash as we explicitly fetch it.
-  SMITHY_TS_COMMIT: "05b2fb2f92af20e824cc71ed4490ed0c7075964e",
+  SMITHY_TS_COMMIT: "25726c12c40479b73c2919c56e2ce8c41e2a22d6",
```
Comment (kstich) [inline scripts/generate-clients/config.js:4]: https://github.com/smithy-lang/smithy-typescript/compare/05b2fb2f92af20e824cc71ed4490ed0c7075964e...25726c12c40479b73c2919c56e2ce8c41e2a22d6

**Review by siddsriv — APPROVED** [review]

File: scripts/generate-clients/config.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // Update this commit when taking up new changes from smithy-typescript.
 module.exports = {
   // Use full commit hash as we explicitly fetch it.
-  SMITHY_TS_COMMIT: "05b2fb2f92af20e824cc71ed4490ed0c7075964e",
```
Comment (trivikr) [inline scripts/generate-clients/config.js:4]: The commit `25726c12c40479b73c2919c56e2ce8c41e2a22d6` is from `bump-0.21.1` branch. We need to bump from merged commit in mainline, which is `cf9d6fe3483ba027f98e87d1921c32dd94e2c1f9`

**Review by trivikr — CHANGES_REQUESTED** [review]

File: scripts/generate-clients/config.js (hunk 2 above)
Comment (trivikr) [reply scripts/generate-clients/config.js:4]: The `export type` change was merged in https://github.com/aws/aws-sdk-js-v3/pull/6171

**Comment by trivikr** [comment]
Merged in https://github.com/aws/aws-sdk-js-v3/pull/6172

**Pull request closed without merging**